Cargo.toml

```diff
@@ -100,6 +100,9 @@ odbc-api = { version = "0.36", optional = true }
 # Faster hashing
 ahash = "0.8"
 
+# workaround for pre-polonius borrow checker
+polonius-the-crab = "0.3"
```

Comment:
I am really not sure about adding a new dependency for this.

Reply:
Removed the dependency but that requires a line of unsafe code (however that crate sort of does the same anyway). It's a borrow-checker issue/bug where the mutable borrow proliferates past the return point, so nothing unsound there.